src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage/index.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: src/layouts/Default.astro
+pubDate: 2024-04-29
+modDate: 2024-04-29
+title: Storage
+description: How to configure storage for a Kubernetes agent
+navOrder: 10
+---
+
+During a deployment, Octopus Server first sends any required scripts and packages to [Tentacle](https://octopus.com/docs/infrastructure/deployment-targets/tentacle) which writes them to the file system. The actual script execution then takes place in a different process called [Calamari](https://github.com/OctopusDeploy/Calamari), which retrieves the scripts and packages directly from the file system. 
+
+On a Kubernetes agent, scripts are executed in separate Kubernetes pods (script pod) as opposed to in a local shell (Bash/Powershell). This means the Tentacle pod and script pods don’t automatically share a common file system.
+
+Since the Kubernetes agent is built on the Tentacle codebase,  it is necessary to configure shared storage so that the Tentacle Pod can write the files in a place that the Script Pods can read from.
+
+We offer two options for configuring the shared storage - you can use either the default NFS storage or specify a Storage Class name.
+
+
+## NFS storage
+
+By default, the Kubernetes agent Helm chart will set up an NFS server suitable for use by the agent inside your cluster. The server runs as a `StatefulSet` in the same namespace as the Kubernetes agent, and uses `EmptyDir` storage, as the working files of the agent are not required to be long-lived. 
+
+This NFS server is referenced in the `StorageClass` that the Kubernetes agent and the script pod use. This `StorageClass` will then instruct the `NFS CSI Driver` to mount the server as directed.
+
+This default implementation is made to let you try the Kubernetes agent without worrying about installing a `ReadWriteMany` compatible `StorageClass` yourself. There are  some drawbacks to this approach:
+
+### Reliability
+Since the NFS server runs inside your Kubernetes cluster, upgrades and other cluster operations will interrupt the server. Due to how NFS stores and allows access to the shared data, clients will not be able to recover from a restart cleanly. This results in deployment failure if one happens to be running while the NFS server is restarted.
```

Comment:
```suggestion
Since the NFS server runs inside your Kubernetes cluster, upgrades and other cluster operations can interrupt cause the NFS server to restart. Due to how NFS stores and allows access to shared data, script pods with an NFS volume mount will not be able to recover cleanly from an NFS server restart. This will result in a deployment failure if one happens to be running when the NFS server is restarted.
```